engine/evd.ml

```diff
@@ -425,6 +427,7 @@ type evar_map = {
   last_mods  : Evar.Set.t;
   (** Metas *)
   metas      : clbinding Metamap.t;
+  evar_flags : evar_flags;
```

Comment:
Rather than a set at the level of evar maps, I think it is probably slightly better to use a flag in the evarinfo. Depending on the kind of uses, this would be more efficient. Also, you could pack the obligation and unresolvable flags into only one integer field. Allocation of evar-related data is quite sensitive, you can easily make compcert or VST go crazy if you're not careful.

Reply:
Using this solution avoids polluting the evar_infos of everything. For unresolvability the next step is to actually replace the evar set of unresolvables with a set of typeclass evars to be resolved (setting it to empty to ensure nested resolutions don't fire the same resolution again). For restricted you might be right, it could help performance. I'd like to see numbers for this solution first.

Reply:
My understanding of the VST slowdown was that we were testing if a big evar map of undefined evars had resolvable ones, which will now result in a lookup in the unresolvable_evars set (which should contain all the goals) instead of testing an evar_extra field. So this could definitely have some impact. It might require inverting the logic to record only the resolvable goals, which should be a small set in general.

Reply:
Indeed, the bench shows a large impact on vst, I'll refine the proposal.